lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -725,7 +725,11 @@ def on_tls_conf_set(
 
         # In case of renewal of the unit transport layer cert - restart opensearch
         if renewal and self.is_admin_user_configured() and self.tls.is_fully_configured():
-            self.tls.reload_tls_certificates()
+            try:
+                self.tls.reload_tls_certificates()
+            except OpenSearchHttpError:
+                logger.error("Could not reload TLS certificates via API, will restart.")
+                self._restart_opensearch_event.emit()
```

Comment:
@reneradoi is this the original TLS flow if the HTTP request fails?

Also, given a restart is expensive, I think it would be worthy here to have a retry.

Reply:
Yes, in case of error, the original restart is initiated. See also here: https://github.com/canonical/opensearch-operator/pull/380/files#diff-5c0d89f838ffd5a9ecf2288b3698320fda2d2b2b737361b6e10adb55d1929f43L732

I've added the retries.